postgres/datadog_checks/postgres/util.py

```diff
@@ -841,3 +841,35 @@ def get_list_chunks(lst, n):
         {'name': 'postgresql.subscription.sync_error', 'type': 'monotonic_count'},
     ],
 }
+
+# Requires PG16+
+STAT_IO_METRICS = {
+    'name': 'stat_io_metrics',
+    'query': """
+SELECT backend_type,
+       object,
+       context,
+       reads,
+       read_time,
+       writes,
+       write_time,
+       extends,
+       extend_time,
+       fsyncs,
+       fsync_time
+FROM pg_stat_io
```

Comment:
Do we need a limit?

Reply:
I don't think so but happy to add one if we think it's warranted for safety. I don't see limit's on the other `pg_stat_*` queries in this file though. 

There should be a reasonably finite number of rows returned here. From the [docs](https://www.postgresql.org/docs/current/monitoring-stats.html#MONITORING-PG-STAT-IO-VIEW) it states
> The pg_stat_io view will contain one row for each combination of backend type, target I/O object, and I/O context, showing cluster-wide I/O statistics. Combinations which do not make sense are omitted.

There's currently 30 rows found when I run this locally
```
datadog_test=> select count(*) from pg_stat_io;                
 count 
-------
    30

```

Reply:
If we trust the documentation, it looks like the worst possible case is: backend_type (12) * object (2) * context (4) = 96.

So no limit is ok (again if we trust the documentation and Postgres has no bugs). `LIMIT 100` with a comment can't hurt though!

> I don't see limit's on the other pg_stat_* queries in this file though.
Well I wouldn't assume that's necessarily for a good reason. Personally I believe that every query we issue should have a limit, but I haven't officially proposed that yet so don't consider it blocking.

Reply:
yeah that sounds like the worst case, but the docs also mention not all combinations exist so I expect it to be less. Regardless I agree that limits are helpful and safer in general and a good habit to get into. I added a `LIMIT 200` and comment as suggested